ga4gh/avrotools.py

```diff
@@ -129,309 +92,419 @@ def _ensureProtocolSubclass(self):
-        """
-        keyGen = cls._stringGenerator()
-        while True:
-            keys = list(itertools.islice(keyGen, size))
-            values = list(itertools.islice(valueGen, size))
-            yield dict(zip(keys, values))
-
-    @classmethod
-    def _recordGenerator(self, recordSchema):
-        """
-        Creates a generator which returns a dictionary where keys and values
-        are determined by the recordSchema. The recordSchema is a map of the
-        records keys to a generator of valid values for that key.
-        """
-        while True:
-            record = {}
-            for key, gen in recordSchema.iteritems():
-                record[key] = gen.next()
-            yield record
+    def handleFixed(self, schema):
+        return self.handleString(schema.size)
 
-    def __init__(self, class_):
-        self.class_ = class_
-        self.schema = class_.schema
+    def handleEnum(self, schema):
+        return random.choice(schema.symbols)
+
+    def handleArray(self, schema, length=10):
+        toReturn = []
+        for _ in xrange(length):
+            item = self.handleSchema(schema.items)
+            toReturn.append(item)
+        return toReturn
+
+    def handleMap(self, schema, size=10):
+        toReturn = {}
+        for _ in xrange(size):
+            key = self.handleString()
+            value = self.handleSchema(schema.values)
+            toReturn[key] = value
+        return toReturn
+
+    def handleUnion(self, schema):
+        chosenSchema = random.choice(schema.schemas)
+        return self.handleSchema(chosenSchema)
 
+    def handleRecord(self, schema):
+        record = {}
+        for field in schema.fields:
+            record[field.name] = self.handleSchema(field.type)
+        return record
+
+
+class TypicalInstanceCreator(AvroTypeSwitch):
+    """
+    Generates typical instances and values
+    """
+    def handleNull(self):
+        raise SchemaToolException()
+
+    def handleBoolean(self):
+        return True
+
+    def handleString(self):
+        return 'aString'
+
+    def handleBytes(self):
+        return 'someBytes'
+
+    def handleInt(self):
+        return 5
+
+    def handleLong(self):
+        return 6
+
+    def handleFloat(self):
+        return 7.0
+
+    def handleDouble(self):
+        return 8.0
+
+    def handleFixed(self, schema):
+        return 'x' * schema.size
+
+    def handleEnum(self, schema):
+        return schema.symbols[0]
+
+    def handleArray(self, schema):
+        val1 = self.handleSchema(schema.items)
+        val2 = self.handleSchema(schema.items)
+        arr = [val1, val2]
+        return arr
+
+    def handleMap(self, schema):
+        value = self.handleSchema(schema.values)
+        mapping = {'key': value}
+        return mapping
+
+    def handleUnion(self, schema):
+        for memberSchema in schema.schemas:
+            if memberSchema.type == 'null':
+                continue
+            return self.handleSchema(memberSchema)
+
```

Comment:
This is definitely dead puppy time... Why not
```python
for memberSchema in schema.schemas:
    if memberSchema.type != 'null':
         return self.handleSchema(memberSchema)
```
This is much clearer to me, and one less line of code. I think the semantics should be documented here: we return an instance of the first non-null schema within the union.